Develop fw 708 2sfk50 40msTS

jmmunoz86 opened the pull request:
this provides the timeslot timing constants for the 2-FSK 50 kbps PHY. 
TS is 40 ms long.

jmmunoz86 committed:
using 2fsk-50 kbps

jmmunoz86 committed:
update config 2-fsk 50kbps

jmmunoz86 committed:
def TS 40 ms

jmmunoz86 committed:
using 2-fsk 50 kbps

jmmunoz86 committed:
using only sub-GHz interface

jmmunoz86 committed:
TS 40 ms

jmmunoz86 committed:
modif slotframe length to 31 TS

jmmunoz86 committed:
using radiotype = 1 (sub-GHz)

jmmunoz86 committed:
update project 40 ms TS

jmmunoz86 committed:
timing for 2fsk50kbps.

jmmunoz86 committed:
2fsk-50 CH with 2 ch

jmmunoz86 committed:
optimize timeslot 2fsk-50 kbps

jmmunoz86 committed:
timeslot of 40 ms. One PHY, subGHz.

jmmunoz86 committed:
2fsk timing OK.

changtengfei commented on openstack/02a-MAClow/IEEE802154E.c:
```diff
@@ -1164,7 +1172,7 @@ port_INLINE void activity_ti1ORri1(void) {
 
 port_INLINE void activity_ti2(void) {
-
+    debugpins_frame_toggle();
```
shouldn't toggle this pin here.

changtengfei commented on openstack/02a-MAClow/IEEE802154E.c:
```diff
@@ -1209,6 +1217,7 @@ port_INLINE void activity_ti2(void) {
     ieee154e_vars.radioOnThisSlot=TRUE;
     // change state
     changeState(S_TXDATAREADY);
+    debugpins_frame_toggle();
```
same here

changtengfei commented on openstack/02a-MAClow/IEEE802154E.h:
```diff
@@ -17,8 +17,12 @@
+//    5,6,12,7,15,4,14,11,8,0,1,2,13,3,9,10
+//};
+
 static const uint8_t chTemplate_default[] = {
```
This channelhopping sequence should move to radio_24ghz/radio_subghz.c file

changtengfei commented on openstack/02a-MAClow/IEEE802154E.h:
```diff
@@ -38,9 +42,9 @@ static const uint8_t ebIEsBytestream[] = {
-#define NUM_CHANNELS                16 // number of channels to channel hop on
-#define DEFAULT_CH_SPACING           1200 // default channel spacing for subghz
-#define DEFAULT_FREQUENCY_CENTER     863625 // defualt freque
+#define NUM_CHANNELS                3 // number of channels to channel hop on
```
num of channels should move to radio part as well

changtengfei commented on openstack/02a-MAClow/IEEE802154E.h:
```diff
@@ -38,9 +42,9 @@ static const uint8_t ebIEsBytestream[] = {
-#define DEFAULT_CH_SPACING           1200 // default channel spacing for subghz
-#define DEFAULT_FREQUENCY_CENTER     863625 // defualt freque
+#define NUM_CHANNELS                3 // number of channels to channel hop on
+#define DEFAULT_CH_SPACING          200 // default channel spacing for subghz
```
this variable could be change if using different phy layers for subghz

changtengfei commented on openstack/02a-MAClow/IEEE802154E.h:
```diff
@@ -38,9 +42,9 @@ static const uint8_t ebIEsBytestream[] = {
-#define DEFAULT_FREQUENCY_CENTER     863625 // defualt freque
+#define NUM_CHANNELS                3 // number of channels to channel hop on
+#define DEFAULT_CH_SPACING          200 // default channel spacing for subghz
+#define DEFAULT_FREQUENCY_CENTER    863125 // defualt freque
```
Same as comment as above

changtengfei commented on openstack/02a-MAClow/IEEE802154E.h:
```diff
@@ -182,12 +186,18 @@ enum ieee154e_atomicdurations_enum {
    wdAckDuration             =   80,                  //  2400us (measured 1000us)
 #else
-   wdAckDuration             =   98,                  //  3000us (measured 1000us)
+   wdAckDuration             =  260,                  //  5400us using 50 kbps
```
This timing variable and above are related to the PHY layer used. So they are no longer belong to IEEE802.15.4E.h file. Define those variables in radio.h files, those variables will be set according which PHY layers are used.

By fixing the length of packet to transmit: 
- For wdDataDuration wdAckDuration, delayTx, delayRx timing variables, they are related to the sending rate of each PHY layer.
- For other timing but slotDuration, they are related the speed to access radio chip
- For slotduration, it's determined both sending rate and the speed to access radio chip. So decide it at last.

changtengfei commented on openstack/02b-MAChigh/schedule.h:
```diff
@@ -17,7 +17,7 @@
 The superframe reappears over time and can be arbitrarily long.
 */
-#define SLOTFRAME_LENGTH    101 //should be 101
+#define SLOTFRAME_LENGTH    31 //should be 101
```
Does it have to be 31?

changtengfei commented on projects/openmote-b/03oos_openwsn/03oos_openwsn.ewt:
```diff
@@ -9,9 +9,9 @@
         <settings>
             <name>C-STAT</name>
-            <archiveVersion>260</archiveVersion>
+            <archiveVersion>261</archiveVersion>
```
ewt file is not needed to be committed, please delete it from repository.

changtengfei commented on openstack/02b-MAChigh/sixtop.c:
```diff
@@ -631,7 +631,7 @@ owerror_t sixtop_send_internal(
 //    }
 
-    msg->l2_radioType = 0;
+    msg->l2_radioType = 1;
```
This is a hack for debugging purpose, you can do this for now., It's better to think what the right way to determine which radioType to use.

changtengfei commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -59,36 +59,36 @@
+    #endif
+    #define PORT_delayRx                         0    //    0us (can not measure)
+    // radio watchdog
+#else // 40 ms
```
The timing below depends on which radio is used by openmote-b, those timing should be moved to radio.h files can call dynamically. 

I recommend to use  a fixed timing for FSK modulation, and use the timing variable for 2.4GHz and OQPSK 800kb/s as well. Only delayTx timing need to be adjust to each three PHY layers. With this, the timing part would be much easier.

changtengfei reviewed (changes requested):
I left comments in the PR, please revise them.

jmmunoz86 committed:
refactoring: changing radiotype for modem.
radiotype defines PHYs whereas modem either sub-ghz or 2.4 ghz.

jmmunoz86 committed:
refactoring and cleaning

changtengfei commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -59,36 +59,38 @@
+    #define PORT_maxRxDataPrepare               30    // 1007us (measured  84us)
+    #define PORT_maxTxAckPrepare                40    //  305us (measured 219us)
+    // radio speed related
+    #define delayTx_2FSK_50                     67
```
The delayTx , delayRx are defined in radio_2d4ghz / radio_subghz files

changtengfei commented on bsp/boards/openmote-b/board_info.h:
```diff
@@ -106,16 +108,32 @@
+   MODEM_2D4GHZ          = 0,
+   MODEM_SUBGHZ          = 1,
+   FREQBAND_ANY          = 2
+} modem_t;
```
why create a modem type here? what the difference betwen this and the radioType

changtengfei commented on bsp/boards/openmote-b/radio_2d4ghz.c:
```diff
@@ -71,7 +71,6 @@ void radio_2d4ghz_setFunctions(radio_functions_t* funcs){
     funcs->radio_rfOff_cb              = radio_2d4ghz_rfOff;
     funcs->radio_setFrequency_cb       = radio_2d4ghz_setFrequency;
     funcs->radio_change_modulation_cb  = radio_2d4ghz_change_modulation;
-    funcs->radio_change_size_cb        = radio_2d4ghz_change_size;
```
if this callback function is not used, you need remove it from the openradio.h file as well.

Also the radio_2d4ghz_change_size function should be removed

changtengfei commented on bsp/boards/openmote-b/radio_subghz.c:
```diff
@@ -46,9 +46,15 @@ typedef struct {
 //=========================== public ==========================================
 static void radio_subghz_read_isr(void);
 static void radio_subghz_clear_isr(void);
+void config_ofdm_1_800_subGHz(void);
```
function should start with name radio_subghz_

changtengfei commented on bsp/boards/openmote-b/radio_subghz.c:
```diff
@@ -46,9 +46,15 @@ typedef struct {
 
 radio_subghz_vars_t radio_subghz_vars;
 
+phy_tsch_config_t              phy_tsch_config_2fsk_50_subGHz;
```
are those variables used in this PR?

changtengfei commented on bsp/boards/openmote-b/radio_subghz.c:
```diff
@@ -412,6 +421,28 @@ uint8_t radio_subghz_calculateFrequency(uint8_t channelOffset, uint8_t asnOffset
+    phy_tsch_config_ofdm_1_800_subGHz.num_channels    = 5;
+
+}
+
```
Here is initialize the variables, when are they being used?

changtengfei commented on bsp/chips/at86rf215/at86rf215.h:
```diff
@@ -3978,30 +4002,33 @@ void at86rf215_readBurst(uint16_t reg, uint8_t* regValueRead, uint16_t size);
-static const registerSetting_t basic_settings_fsk_option1 []={
+
+
+static const registerSetting_t basic_settings_fsk_option1_subghz []={
```
why rename it?

changtengfei commented on projects/openmote-b/03oos_openwsn/03oos_openwsn.ewd:
```diff
@@ -88,7 +88,7 @@
                 <option>
                     <name>OCLastSavedByProductVersion</name>
-                    <state>8.11.1.13270</state>
+                    <state>8.20.1.14181</state>
```
the project file changes shouldn't commit

changtengfei commented on projects/openmote-b/03oos_macpong/03oos_macpong.ewp:
```diff
@@ -66,7 +66,7 @@
                 <option>
                     <name>OGLastSavedByProductVersion</name>
-                    <state>8.11.1.13270</state>
+                    <state>8.20.1.14181</state>
```
the project file changes like version shouldn't commit.

changtengfei reviewed (changes requested):
Please don't add new changes after the review. I left some comments, please just fix them, no more new changes added.

jmmunoz86 committed:
adding timing for 3 PHYs

jmmunoz86 committed:
corrected TsTxOffset. SPI config happens at the beginning of each TS.

jmmunoz86 committed:
changed startup_iar.c, flashing with IAR and scons both working

jmmunoz86 committed:
OpenWSN working with only OFDM1, TS 40 ms

jmmunoz86 committed:
modifies the open_addr_t struct.
Creates the addr_16b_t and addr_64b_t, containing the addresses and the radioType item.

jmmunoz86 committed:
Merge branch 'develop_fw-708-2fsk_OFDM1' into develop_FW-708_2sfk50_40ms

jmmunoz86 committed:
Slotframe of 33 TS. faster synch.

jmmunoz86 committed:
Merge branch 'develop_FW-708_OFDM1_40ms' into develop_FW-708_2sfk50_40ms

# Conflicts:
#	bsp/boards/openmote-b/board_info.h

jmmunoz86 committed:
added radioType to neighbors.
Same device with X radios is seen as X neighbors.

jmmunoz86 committed:
return parent address and radioType

jmmunoz86 committed:
assign the radioType on which the 6p transaction will take place.
use a different radioType for consecutive EBs.
If radioType is 0 (2.4ghz), make sure it is used radioType 1 (OFDM).

jmmunoz86 committed:
Not nice workaround to use a different radioType for consecutive DIOs.
Consider the radioType when updating the rank in the neighbor table.

jmmunoz86 committed:
send DAO on the correct radioType

jmmunoz86 committed:
Increase maxTxDataPrepare to allow long packets to be loaded into the radio.

jmmunoz86 committed:
casting to correct type (radioType)

jmmunoz86 committed:
MAX output power

jmmunoz86 committed:
EB sent on each PHY

jmmunoz86 committed:
send DIO on each radioType available

jmmunoz86 committed:
SLOTFRAME_LENGTH of 101 TS

jmmunoz86 committed:
adding radioType criterion to neighbors

jmmunoz86 committed:
Configuring the front end radio at synch time.

jmmunoz86 committed:
function to read the state of the radio.
AT86RF215 goes to TXPREP after receiving a frame.
After receiving a frame, putting the radio back to listen.

jmmunoz86 committed:
delayTX of OFDM (synch with this PHY).
radio keeps listening despite the RXFE ISR.
No need to reload the RXFS and RXFE callbacks.

changtengfei commented:
Can one of the admins verify this patch?
